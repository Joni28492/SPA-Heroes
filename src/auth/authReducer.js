import { types } from "../types/types";

// const state = {
//     name: 'Joni',
//     logged: true,
// }//solo si esta autenticado

export const authReducer = (state = {}, action) => {
    switch (action.type) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            }
        case types.logout:
            return {
                logged: false,
            }//no interesa el payload
    
        default:
            return state;
    }
}


