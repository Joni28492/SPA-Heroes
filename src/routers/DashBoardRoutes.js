

import React from 'react'
import { Redirect, Route, Switch } from 'react-router'
import { DcScreen } from '../components/dc/DcScreen'
import { HeroScreen } from '../components/heroes/HeroScreen'
import { MarvelScreen } from '../components/marvel/MarvelScreen'
import { SearchScreen } from '../components/search/SearchScreen'
import { Navbar } from '../components/ui/NavBar'

export const DashBoardRoutes = () => {
    return (
        <>
            <Navbar />
            <div className="container mt-2">
                <Switch>
                    <Route exact path='/marvel' component={MarvelScreen} />
                    {/* Pasamos argumentos con : */}
                    <Route exact path='/hero/:heroeId' component={HeroScreen} />
                    <Route exact path='/dc' component={DcScreen} />
                    <Route exact path='/search' component={SearchScreen} />

                    <Redirect to="/marvel"/>
                </Switch>
            </div>
        </>
    )
}
