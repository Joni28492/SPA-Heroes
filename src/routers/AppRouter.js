

import React, { useContext } from 'react';
import {
    HashRouter as Router,
    Switch
} from "react-router-dom";
import { AuthContext } from '../auth/AuthContext';
import { LoginScreen } from '../components/login/LoginScreen';
import { DashBoardRoutes } from './DashBoardRoutes';
import { PrivateRouter } from './PrivateRouter';
import { PublicRoute } from './PublicRoute';


export const AppRouter = () => {

    const {user} = useContext(AuthContext);

    return (
        <Router>
            <div>
               
                <Switch>
                    <PublicRoute 
                        exact path="/login" 
                        component={LoginScreen}
                        isAuthenticated={user.logged}
                    />

                    <PrivateRouter  
                        path="/" 
                        component={DashBoardRoutes}
                        isAuthenticated={user.logged}
                    />
                </Switch>
            </div>
        </Router>
    )
}
