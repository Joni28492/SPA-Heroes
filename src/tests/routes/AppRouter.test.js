import React from 'react';
import { mount, shallow } from "enzyme"
import { AppRouter } from '../../routers/AppRouter';
import { AuthContext } from '../../auth/AuthContext';

describe('Pruebas en <AppRouter />', () => {
    const contextValue = {
        dispatch: jest.fn(),
        user: {logged: false}
    }
    test('debe de mostrar el login si no esta autenticado ', () => {
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                {/* tenemos que proveerle un contexto */}
                <AppRouter />
            </AuthContext.Provider>  
        );
        // console.log(wrapper.html())
        expect(wrapper).toMatchSnapshot();
    })
    

    test('debe de mostrar el componente de marvel si esta autenticado', () => {
        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'Joni'
            }
        }
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>  
        );
        expect(wrapper.find('.navbar').exists()).toBe(true)
    })
    
})
