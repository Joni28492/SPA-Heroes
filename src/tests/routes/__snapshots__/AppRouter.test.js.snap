// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe de mostrar el login si no esta autenticado  1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      className="btn  btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
